bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/automation/jrubyscripting/internal/JRubyScriptEngineFactory.java

```diff
@@ -102,7 +105,26 @@ public void scopeValues(ScriptEngine scriptEngine, Map<String, Object> scopeValu
                         .map(JRubyScriptEngineFactory::mapGlobalPresets) //
                         .collect(Collectors.toMap(map -> map.getKey(), map -> map.getValue())); //
 
-        super.scopeValues(scriptEngine, filteredScopeValues);
+        Map<Boolean, Map<String, Object>> partitionedMap = //
+                filteredScopeValues.entrySet() //
+                        .stream() //
+                        .collect(Collectors.partitioningBy(entry -> (entry.getValue() instanceof Class),
+                                Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+
+        importClassesToRuby(scriptEngine, partitionedMap.getOrDefault(true, new HashMap<String, Object>()));
+        super.scopeValues(scriptEngine, partitionedMap.getOrDefault(false, new HashMap<String, Object>()));
```

Comment:
You can use the diamond operator here:

```suggestion
        importClassesToRuby(scriptEngine, partitionedMap.getOrDefault(true, new HashMap<>()));
        super.scopeValues(scriptEngine, partitionedMap.getOrDefault(false, new HashMap<>()));
```

Reply:
fixed